usage example for load_music

SimonRhook opened the pull request:
# Overview
Usage example to demonstrate load music function.

Updated formatting from PR #324 from trimester 3 2024 - note #324 already has 1 approved review

Splashkit Function: `load_music`

# Files Included
- [x] Title and explanation (.txt)
- [x] C++ code (SplashKit)
- [] C++ code (Beyond SplashKit)
- [x] C# code (top-level statements)
- [x] C# code (Object-Oriented Programming)
- [x] Python code
- [x] .webm

# Usage Example Checks
- [x] Simple, clear demonstration of the function
- [x] Code uses Splashkit functions
- [x] Tested in Chrome and Firefox

cameronhum commented on public/usage-examples/audio/load_music-1-example-oop.cs:
```diff
@@ -0,0 +1,28 @@
+        {
+            // Check if audio is ready to use
+            if (!SplashKit.AudioReady())
+                SplashKit.OpenAudio();
```
needs brackets added

cameronhum commented on public/usage-examples/audio/load_music-1-example-oop.cs:
```diff
@@ -0,0 +1,28 @@
+            if (SplashKit.HasMusic("adventure"))
+                SplashKit.WriteLine("Music successfully loaded. Ready for playback.");
+            else
+                SplashKit.WriteLine("Loading music failed.");
```
needs brackets added

cameronhum commented on public/usage-examples/audio/load_music-1-example-sk.cpp:
```diff
@@ -0,0 +1,21 @@
+int main()
+{
+    // Check if audio is ready to use
+    if(! audio_ready())
```
spacing needs fixing

cameronhum commented on public/usage-examples/audio/load_music-1-example.py:
```diff
@@ -0,0 +1,18 @@
+from splashkit import *
+
+
```
double spacing needs to be fixed

cameronhum commented on public/usage-examples/audio/load_music-1-example-sk.cpp:
```diff
@@ -0,0 +1,21 @@
+
+    // Cleanup
+    free_all_music();    
+
```
can delete this extra line

cameronhum commented on public/usage-examples/audio/load_music-1-example-top-level.cs:
```diff
@@ -0,0 +1,19 @@
+
+// Cleanup
+FreeAllMusic();
+
```
can delete this extra line

cameronhum commented on public/usage-examples/audio/load_music-1-example-oop.cs:
```diff
@@ -0,0 +1,28 @@
+namespace LoadMusicExample
+{
+    public class Program
+
```
can delete this extra line

cameronhum commented on public/usage-examples/audio/load_music-1-example.png:
Since the image should represent all the languages. The python3 success statement probably should be left out

cameronhum commented on public/usage-examples/audio/load_music-1-example-oop.cs:
```diff
@@ -0,0 +1,28 @@
+            if (!SplashKit.AudioReady())
+                SplashKit.OpenAudio();
+
+            // Load music file
```
Comment is a bit self explanatory (its same as the function name). could change it a bit

cameronhum reviewed (changes requested):
I have tested this in the site and it loads and the resources are downloadable. This was also tested locally and all code examples work and are consistent in their functionality between them. This is a good simple usage example.

Other than my comments I think something else needs to be added to this to show off its usefulness. From what there is so far, the music is being loaded however isn't used and by running its not proven that the function does anything as we just have to trust its has music output. I think some sort of visual or audio queue to prove its loaded would be good that's different from the other usage example you have already done. However if there's nothing else that can show it without doing similar to the other examples I think this would be fine to leave if necessary. 


![image](https://github.com/user-attachments/assets/f3439240-b66a-4a33-ac77-734e78769768)

# Peer Review

I've reviewed the LoadMusic Usage Example

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
- [x] Code correctly uses Splashkit functions.
- [ ] Code clearly demonstrates the function.
- [ ] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [ ] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview

SimonRhook commented:
I've updated based on your recommendations. I've also added an array with different music names to cycle through and demonstrate the logic for when the music inst loaded. Hopefully this addresses your concern about usefulness

cameronhum reviewed (approved):
# Peer Review

I've reviewed the LoadMusic and It is now much cleaner and I love the functionality you added. It keeps it incredibly simple but feels much more complete! Well done!

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
- [x] Code correctly uses Splashkit functions.
- [x] Code clearly demonstrates the function.
- [x] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [x] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview

JulianBurton reviewed (approved):
I think this is a good example of the Usage example. I think that an improvement could be to add a visual aspect in a window, but I do not thank that it is a necessary enough addition to warrant rebuilding the program to run in a window.

Peer Review
I've reviewed the Usage Example for LoadMusic

# Checks
- [x]  All required files are present.
     - [x]  Example Title (.txt)
     - [x]  C++ code (SplashKit)
     - [x]  C# code (top-level statements)
     - [x]  C# code (Object-Oriented Programming)
     - [x]  Python code
- [x]  Code correctly uses Splashkit functions.
- [x]  Code clearly demonstrates the function.
- [x]  All versions maintain the same structure and comments.
# Code Tests done
- [x]  C++ SplashKit code ran correctly.
- [x]  C++ Beyond SplashKit code ran correctly.
- [x]  C# top level code ran correctly.
- [x]  C# OOP code ran correctly.
- [x]  Python code ran correctly.
# Website Tests done
- [x]  npm run build
- [x]  npm run preview
